debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9IndexableObjectHelper.java

```diff
@@ -120,26 +135,93 @@ public static U32 size(J9ObjectPointer objPointer) throws CorruptDataException
 		return size(J9IndexableObjectPointer.cast(objPointer));
 	}
 
+	/**
+	 * Get the header size of the contiguous indexable object
+	 */
 	public static long contiguousHeaderSize()
 	{
+		try {
+			UDATA isDataAddrPresent = J9RASHelper.getVM(DataType.getJ9RASPointer()).isIndexableDataAddrPresent();
+			if(0 == isDataAddrPresent.intValue()) {
+				return contiguousHeaderSizeNoDataAddress();
+			} else {
+				return contiguousHeaderSizeWithDataAddress();
+			}
+		} catch (NoSuchFieldException | CorruptDataException e) {
+			return (contiguousHeaderSizeNoDataAddress() + UDATA.SIZEOF);
+		}
+	}
+
+	/**
+	 * Get the header size of the contiguous indexable object without the dataAddress header field
+	 */
+	public static long contiguousHeaderSizeNoDataAddress()
+	{
+		if (mixedReferenceMode) {
+			if (compressObjectReferences) {
+				return (J9IndexableObjectContiguousCompressed.SIZEOF);
+			}
+			return (J9IndexableObjectContiguousFull.SIZEOF);
+		}
+		return (J9IndexableObjectContiguous.SIZEOF);
+	}
+
+	/**
+	 * Get the header size of the contiguous indexable object with the dataAddress header field
+	 */
+	public static long contiguousHeaderSizeWithDataAddress() {
 		if (mixedReferenceMode) {
 			if (compressObjectReferences) {
-				return J9IndexableObjectContiguousCompressed.SIZEOF;
+				return J9IndexableObjectWithDataAddressContiguousCompressed.SIZEOF;
 			}
-			return J9IndexableObjectContiguousFull.SIZEOF;
+			return J9IndexableObjectWithDataAddressContiguousFull.SIZEOF;
 		}
-		return J9IndexableObjectContiguous.SIZEOF;
+		return J9IndexableObjectWithDataAddressContiguous.SIZEOF;
 	}
 
+	/**
+	 * Get the header size of the discontiguous indexable object
+	 */
 	public static long discontiguousHeaderSize()
+	{
+		try {
+			UDATA isDataAddrPresent = J9RASHelper.getVM(DataType.getJ9RASPointer()).isIndexableDataAddrPresent();
+			if(0 == isDataAddrPresent.intValue()) {
+				return discontiguousHeaderSizeNoDataAddress();
+			} else {
+				return discontiguousHeaderSizeWithDataAddress();
+			}
+		} catch (NoSuchFieldException | CorruptDataException e) {
+			return (discontiguousHeaderSizeNoDataAddress() + UDATA.SIZEOF);
+		}
+	}
+
+	/**
+	 * Get the header size of the discontiguous indexable object without the dataAddress header field
```

Comment:
Please finish each sentence with a period.

Reply:
Please don't add periods to `@param` or `@throws` clauses, and documentation should have more than just those clauses.